src/components/tabs/TabGroup.svelte

```diff
@@ -1,10 +1,13 @@
 <script>
   export let active = 0;
   import { writable } from "svelte/store";
-  import { setContext } from "svelte";
+  import { createEventDispatcher, setContext } from "svelte";
 
-  const store = writable(active);
-  setContext("activeTab", store);
+  const activeTab = writable(active);
+  setContext("activeTab", activeTab);
+
+  const dispatch = createEventDispatcher();
+  $: dispatch("tabChanged", { active: $activeTab });
```

Comment:
This seems like an overkill: we're using 3 separate state management systems — store, context, event dispatcher — to handle active tab state. Is it possible to reduce the complexity here?